frontend/Auth/Login.tsx

```diff
@@ -1,9 +1,165 @@
+
+[x] If the backend responds with a 200, it can be assumed that a cookie was set and the user is now logged in. At that time, the user should be redirected back to the home page.
+*/
+
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import { Button, Form, Card } from "semantic-ui-react";
+import { RouterProps } from "react-router";
+import { useMutation } from "react-query";
+import { flaxFetch } from "../Utils/flaxFetch";
+import { unary } from "lodash-es";
+
+export function Login(props: RouterProps) {
+  const [loginFormData, setLoginFormData] = useState<LoginFormData>({
+    username: "",
+    password: "",
+  });
+
+  const history = useHistory();
+
+  const submitMutation = useMutation<
+    LoginResultData,
+    Error,
+    React.FormEvent<HTMLFormElement>
+  >(
+    (evt) => {
+      evt.preventDefault();
+      let requestPromise = flaxFetch<LoginResultData>(`/login`, {
+        method: "POST",
+        body: {
+          username: loginFormData.username,
+          password: loginFormData.password,
+        },
+      });
+      return requestPromise;
+    },
+    {
+      onSuccess: async (data, variables, context) => {
+        history.push("/home");
+      },
+    }
+  );
 
-export function Login(props) {
   return (
-    <Form method="post" action="/login">
-      <Button type="submit">Login</Button>
-    </Form>
+    <div style={{ display: "flex", justifyContent: "center" }}>
+      <Card
+        style={{
+          padding: "10px",
+          width: "30%",
+          boxShadow: "none",
+          marginTop: "15%",
+        }}
+      >
+        <h1 style={{ color: "#646461", fontSize: "20px" }}>Sign in</h1>
+        <Form onSubmit={unary(submitMutation.mutate)}>
+          <Form.Field>
+            <label htmlFor="username">Email</label>
+            <input
+              id="username"
+              value={loginFormData.username}
+              onChange={(evt) =>
+                setLoginFormData({
+                  ...loginFormData,
+                  username: evt.target.value,
+                })
+              }
+              required
+            />
+          </Form.Field>
+
+          <Form.Field>
+            <label htmlFor="password">Password</label>
+            <input
+              id="password"
+              value={loginFormData.password}
+              onChange={(evt) =>
+                setLoginFormData({
+                  ...loginFormData,
+                  password: evt.target.value,
+                })
+              }
+              required
+            />
+          </Form.Field>
+
+          <Button
+            type="submit"
+            style={{
+              background: "#2a467b",
+              color: "white",
+              width: "50%",
+              margin: 0,
+            }}
+          >
+            Sign in
+          </Button>
+
+          <Button
+            type="submit"
```

Comment:
I don't think this should be `<Button type="submit">`, but rather `<Link href="/reset-password">`, since the reset password page will be a separate route